test/widgets/actions_test.dart

```diff
@@ -56,6 +56,23 @@ void main() {
     }
 
     group('markNarrowAsRead', () {
+      Future<void> confirmToMarkNarrowAsRead({required WidgetTester tester,
+        required BuildContext context, required Narrow narrow, required int unreadCount,
+      }) async {
+        final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
+        final future = tester.runAsync(() => ZulipAction.markNarrowAsRead(context, narrow));
+
+        await tester.pump();
+        final (confirmButton, _) = checkSuggestedActionDialog(tester,
+          expectedTitle: zulipLocalizations.markAllAsReadConfirmationDialogTitle,
+          expectedMessage: zulipLocalizations.markAllAsReadConfirmationDialogMessage(unreadCount),
+          expectedActionButtonText: zulipLocalizations.markAllAsReadConfirmationDialogAction,
+        );
+        await tester.tap(find.byWidget(confirmButton));
+        await tester.pump();
+        await future;
```

Comment:
The use of `tester.runAsync` here looks unusual. How about this:

```suggestion
        final zulipLocalizations = GlobalLocalizations.zulipLocalizations;

        final future = ZulipAction.markNarrowAsRead(context, narrow);
        await tester.pump(); // confirmation dialog appears
        final (confirmButton, _) = checkSuggestedActionDialog(tester,
          expectedTitle: zulipLocalizations.markAllAsReadConfirmationDialogTitle,
          expectedMessage: zulipLocalizations.markAllAsReadConfirmationDialogMessage(unreadCount),
          expectedActionButtonText: zulipLocalizations.markAllAsReadConfirmationDialogConfirmButton,
        );
        await tester.tap(find.byWidget(confirmButton));
        await tester.pump(Duration.zero); // wait through API request
        await future;
```

Reply:
Also, let's factor this differently: I like the `checkConfirmDialog` helper you've added in test/widgets/message_list_test.dart; let's follow that pattern.

To do that, you could copy-paste my suggestion above, then inline all of this helper `confirmToMarkNarrowAsRead`, then factor out just the `checkConfirmDialog` part.